src/hotspot/share/services/heapDumper.cpp

```diff
@@ -787,6 +789,91 @@ class DumperSupport : AllStatic {
   }
 };
 
+// Hash table of klasses to the klass metadata. This should greatly improve the
+// hash dumping performance. This hash table is supposed to be used by a single
+// thread only.
+//
+class DumperClassCacheTableEntry : public CHeapObj<mtServiceability> {
+  friend class DumperClassCacheTable;
+private:
+  GrowableArray<char> _sigs_start;
+  GrowableArray<int> _offsets;
```

Comment:
These arrays are allocated in the resource area so don't we need a resource mark somewhere around the usage of `DumperClassCacheTable` which creates these `DumperClassCacheTableEntry` objects?

Reply:
Oh, never mind. I think I've reviewed an outdated version and you've added the resource mark with https://github.com/openjdk/jdk/pull/16545/commits/a19501f9df003c9370af037d12abcce352c7986f